src/data/nat/factorization.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2021 Stuart Presnell. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stuart Presnell
+-/
+import data.nat.prime
+import data.nat.mul_ind
+
+/-!
+# Prime factorizations
+
+ `n.factorization` is the finitely supported function `ℕ →₀ ℕ`
+ mapping each prime factor of `n` to its multiplicity in `n`.  For example, since 2000 = 2^4 * 5^3,
+  * `factorization 2000 2` is 4
+  * `factorization 2000 5` is 3
+  * `factorization 2000 k` is 0 for all other `k : ℕ`.
+-/
```

Comment:
As discussed [here on Zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there.20code.20for.20X.3F/topic/Multiplicity.20in.20the.20naturals) we should move many of the `factors.count` and `padic_val_nat` results to this file, and have the existing ones as consequences. For the sake of this PR it can be a TODO if you prefer.